### PR TITLE
fix(parallel_names): first fix of bug

### PR DESCRIPTION
**Pull request by Tanguylo**

### CODE REVIEW

**Comment by Tanguylo** [comment]
Auto adapt fontsize considering available space in a 2D bounding box.
Can still be improved by adding a small but complicated algorithm that i think is not required for the moment.
If you need a test on your data @BayssetGuillaume , please provide me the html and i'll send you equivalent pics.

image

image

**Review by GhislainJ — CHANGES_REQUESTED** [review]

File: src/plot-data.ts [hunk 1]
```diff
@@ -841,18 +845,52 @@ export abstract class PlotData {
...
+      let align = "center";
+      const textParams: textParams = { width: width, height: origin.y - 2 - this.Y, align: align, baseline: "bottom", multiLine: true };
+      let axisTitle = new newText(this.axis_list[i]['name'], origin, textParams);
+      axisTitle.format(this.context);
+
+      let boxOriginX = 0;
+      let axisLocation = 0;
+      if (i == 0 && axisTitle.width > (this.axis_x_start - this.X) * 2) {
+        axisLocation = -1;
+        const offset = Math.min(axisTitle.width, this.x_step / 2);
```
[the remark this answers is not in the document]
Comment (Tanguylo) [reply src/plot-data.ts:860]: Thanks

File: src/plot-data.ts [hunk 2]
```diff
@@ -841,18 +845,52 @@ export abstract class PlotData {
...
+        axisTitle.origin.x = current_x - offset * 0.95;
+        axisTitle.width =  offset * 0.95 + (this.width - this.axis_x_end + this.X) * 0.9;
+        axisTitle.align = "left";
+      }
+      axisTitle.format(this.context);
+
+      // Weird but vowed to disappear
+      if (axisLocation == 0) {
+        boxOriginX = origin.x - axisTitle.width / 2;
+        axisTitle.format(this.context);
```
[the remark this answers is not in the document]
Comment (Tanguylo) [reply src/plot-data.ts:876]: You are pointing out that i did not write a while loop in newText to get the best font size for a given rectangle

File: src/plot-data.ts [hunk 3]
```diff
@@ -841,18 +845,52 @@ export abstract class PlotData {
...
+      } else if (axisLocation == -1) {
+        boxOriginX = axisTitle.origin.x - axisTitle.width;
+      } else if (axisLocation == 1) {
+        boxOriginX = axisTitle.origin.x;
+        axisTitle.format(this.context);
```
[the remark this answers is not in the document]
Comment (Tanguylo) [reply src/plot-data.ts:881]: Same answer ;)

File: src/plot-data.ts [hunk 4]
```diff
@@ -2968,25 +3034,38 @@ export class Interactions {
...
+      if (plot_data.context.isPointInPath(plot_data.axisNamesBoxes[i].path, mouse1X, mouse1Y)) {
         selected_name_index = i;
+        click_on_name = true;
         break;
       }
     }
     return [click_on_name, selected_name_index];
   }
 
+  // public static initialize_click_on_name(nb_axis:number, mouse1X:number, mouse1Y:number, plot_data:any) {
```
[the remark this answers is not in the document]
Comment (Tanguylo) [reply src/plot-data.ts:3046]: Removed

File: src/subplots.ts [hunk 5]
```diff
@@ -249,6 +249,7 @@ export class ParallelPlot extends PlotData {
       super(data, width, height, buttons_ON, X, Y, canvas_id, is_in_multiplot);
       if (!is_in_multiplot) {
         var requirement = '0.6.1';
+        console.log(data, data.package_version)
```
[the remark this answers is not in the document]
Comment (Tanguylo) [reply src/subplots.ts:252]: Yesssss !

File: src/plot-data.ts [hunk 6]
```diff
@@ -928,18 +967,45 @@ export abstract class PlotData {
...
+
+      let origin = new Vertex(this.axis_x_start, current_y + 10);
+      const textParams: textParams = { width: this.width * 0.25, height: this.y_step * 0.98, align: "center", baseline: "hanging", multiLine: true };
+      let axisTitle = new newText(this.axis_list[i]['name'], origin, textParams);
+
+      axisTitle.format(this.context);
+
+      let control = false;
+      let standard = true;
+      if (axisTitle.width > (this.axis_x_start - this.X) * 2) {
```
[the remark this answers is not in the document]
Comment (Tanguylo) [reply src/plot-data.ts:979]: Called it RIGHT_SPACE

**Review by GhislainJ — APPROVED** [review]

**Pull request merged**